src/logic/calculate.js

```diff
@@ -0,0 +1,105 @@
+import operate from './operate';
+
+const calculate = (dataObject, buttonName) => {
+  let { total, next, operation } = dataObject;
+
+  const ArithmeticOperators = ['+', '-', 'X', '÷'];
+
+  const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '.'];
+  const modifierOrPercent = ['+/-', '%'];
+  const modifierOrPercentFunc = (number, operator) => {
+    if (next) {
+      if (operation && !total) {
+        operation = null;
+      }
+      next = operate(next, number, operator);
+    } else if (total) {
+      operation = null;
+      total = operate(total, number, operator);
+    }
+  };
+
+  if (buttonName === 'AC') {
+    total = null;
+    next = null;
+    operation = null;
+  } else if (modifierOrPercent.includes(buttonName)) {
+    if (buttonName === '+/-') {
+      modifierOrPercentFunc(-1, 'X');
+    } else if (buttonName === '%') {
+      modifierOrPercentFunc(100, '÷');
+    }
+  } else if (buttonName === '=') {
+    if (total && next && operation) {
+      total = operate(total, next, operation);
+      next = null;
+      operation = null;
+    }
+  } else if (ArithmeticOperators.includes(buttonName)) {
+    if (total && next && operation) {
+      total = operate(total, next, operation);
+      next = null;
+      operation = buttonName;
+    } else {
+      operation = buttonName;
+    }
+  } else if (numbers.includes(buttonName)) {
+    if (next) {
+      if (operation) {
+        if (total) {
+          if (buttonName === '.') {
+            if (!next.includes(buttonName)) {
+              next += buttonName;
```

Comment:
This nested `if` is too much. Are you sure there's no alternative way to do this? 🤔

Reply:
I'll check for alternative ways to reduce the nested if statements and make adjustments.